docs/src/main/tut/datatypes/nel.md

```diff
@@ -0,0 +1,182 @@
+
+This is much more precise and we don't have to wonder whether the list
+of errors might be empty when reporting them later on.
+
+### Avoiding `Option` by demanding more specific arguments
+
+As functional programmers, we naturally shy away from partial
+functions that can throw exceptions like the famous `head` method on,
+e.g., `List`.
+
+Let's take as an example a method that calculates the `average`:
+
+```tut:silent
+def average(xs: List[Int]): Double = {
+  xs.sum / xs.length.toDouble
+}
+```
+
+Clearly, this is not a valid definition for *empty* lists, because
+division by zero will throw an exception.  To fix this, one way is to
+return an `Option` instead of a `Double` right away:
+
+```tut:silent
+def average(xs: List[Int]): Option[Double] = if (xs.isEmpty) {
+  None
+} else {
+  Some(xs.sum / xs.length.toDouble)
+}
+```
+
+That works and is safe, but this only masks the problem of *accepting
+invalid input*.  By using `Option`, we extend the `average` function
+with the logic to handle empty lists.  Additionally, all callers have
+to handle the `Option` cases, maybe over and over again.  While better
+than failing with an exception, this is far from perfect.
+
+Instead what we would like to express is that `average` does not make
+sense at all for an empty list.  Luckily, cats defines the
+`NonEmptyList`.  As the name says, this represents a list that
+*cannot*, by construction, be empty.  So given a `NonEmptyList[A]` you
+can be sure that there is at least one `A` in there.
+
+Let's see how that impacts your `average` method:
+
+```tut:silent
+import cats.data.NonEmptyList
+def average(xs: NonEmptyList[Int]): Double = {
+  xs.reduceLeft(_+_) / xs.length.toDouble
+}
+```
+
+With that, `average` is free of any "domain invariant validation" and
+instead can focus on the actual logic of computing the average of the
+list.  This ties in nicely with the recommendation of shifting your
+validation to the very borders of your program, where the input enters
+your system.
+
+## Structure of a NonEmptyList
+
+`NonEmptyList` is defined as follows:
+
+```scala
+final case class NonEmptyList[+A](head: A, tail: List[A]) {
+  // Implementation elided
+}
+```
+
+The `head` of the `NonEmptyList` will be _non-empty_. Meanwhile, the
+`tail` can have zero or more elements contained in a `List`.
+
+## Defined for all its elements
+
+An important trait of `NonEmptyList` is the totality. For `List` specifically,
+both `head` and `tail` are partial: they are only well-defined if it
+has at least one element.
+
+`NonEmptyList` on the other hand, _guarantees_ you that operations like
+`head` and `tail` are defined, because constructing an empty
+`NonEmptyList` is simply not possible!
+
+## Constructing a NonEmptyList
+
+To construct a `NonEmptyList` you have different possibilities.
+
+### Using `one`
+
+If you want to construct a collection with only one argument, use
+`NonEmptyList.one`:
+
+```tut
```

Comment:
I'm not sure that we have been 100% consistent, but I think that most of the time we've been using `tut:book` in (non-silent) examples so they compile when being copy/pasted.